runtime/src/main/java/org/corfudb/util/SamplingUtils.java

```diff
@@ -0,0 +1,53 @@
+package org.corfudb.util;
+
+import static com.carrotsearch.sizeof.RamUsageEstimator.sizeOf;
+
+import java.util.Arrays;
+
+/**
+ * Select a random subset k, from a stream that has size n,
+ * but n is unknown (i.e. online sampling).
+ *
+ * Created by maithem on 8/7/17.
+ */
+public class SamplingUtils {
+
+    private int k;
+    private long n;
+    private long[] sample;
+    private int x;
+
+    public SamplingUtils(int k, long p) {
```

Comment:
Initially, I wanted p to be the percentage of n samples that should be considered, but in this case n is not known ahead of time. I will remove, since we just need a random subset of size k to consider.